### PR TITLE
Port splitCheck and splitCheckObserver implementations, step 1

### PR DESCRIPTION
**Pull request by xiaobogaga**
@coocood PTAL, thanks.

### CODE REVIEW

File: tikv/raftstore/coprocessor.go [hunk 1]
```diff
@@ -50,6 +50,222 @@ type splitChecker interface {
...
+func (observer *keysSplitCheckObserver) stop() {}
+
+func (observer *keysSplitCheckObserver) addChecker(obCtx *observerContext, host *splitCheckerHost, db *badger.DB,
+	policy pdpb.CheckPolicy) {
+	//todo, currently it is just a placeholder
+}
+
+type tableSplitChecker struct {
+	firstEncodedTablePrefix []byte
+	spKeys                  []byte
```
Comment (coocood) [inline tikv/raftstore/coprocessor.go]: splitKey

File: tikv/raftstore/coprocessor.go [hunk 2]
```diff
@@ -50,6 +50,222 @@ type splitChecker interface {
 	policy() pdpb.CheckPolicy
 }
 
+type sizeSplitChecker struct {
+	maxSize         uint64
+	splitSize       uint64
+	currentSize     uint64
+	spKeys          [][]byte
```
Comment (coocood) [inline tikv/raftstore/coprocessor.go]: use `splitKeys` is more consistent.

File: tikv/raftstore/coprocessor.go (hunk 1 above)
Comment (xiaobogaga) [reply tikv/raftstore/coprocessor.go]: ok, please wait.

**Comment by xiaobogaga** [comment]
@coocood PTAL again, thanks.

File: tikv/raftstore/coprocessor.go [hunk 3]
```diff
@@ -225,14 +225,14 @@ func (observer *keysSplitCheckObserver) addChecker(obCtx *observerContext, host
 
 type tableSplitChecker struct {
 	firstEncodedTablePrefix []byte
-	spKeys                  []byte
+	splitKeys               []byte
```
Comment (coocood) [inline tikv/raftstore/coprocessor.go]: `splitKey`

Comment (xiaobogaga) [reply tikv/raftstore/coprocessor.go]: yes, thanks

**Comment by xiaobogaga** [comment]
@coocood PTAL again, thanks

**Comment by coocood** [comment]
LGTM

**Pull request merged**